.github/workflows/post_benchmark_comment.yaml

```diff
@@ -0,0 +1,108 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: Post benchmark comment
+
+on:
+  workflow_run:
+    workflows: ["CI"]
+    types: [completed]
+
+concurrency:
+  # A PR number if a pull request and otherwise the commit hash. This cancels
+  # queued and in-progress runs for the same PR (presubmit) or commit
+  # (postsubmit).
+  group: ${{ github.event.workflow_run.pull_requests[0].number || github.sha }}-post-benchmark-comment
```

Comment:
Huh seems like we should have a suffix in the concurrency group for the ci workflow.

I don't think that `pull_requests[0]` is going to do what you want it to here. I was suspicious because it's a list. The GitHub documentation is supremely unhelpful on this topic, so I just tested it out and the `pull_requests` array for the latest CI run on this PR is empty `gh api --jq '.pull_requests' /repos/iree-org/iree/actions/runs/3878260126`. The GitHub docs give an example of posting a comment to a PR. Unfortunately, they rely on the untrusted workflow itself telling you what PR to post to :'-( https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#using-data-from-the-triggering-workflow. I think that may be the best we can do though. I've looked through the event payload and don't see anything that could get us back to the original event trigger. Ugh GitHub is so annoying :-/

Reply:
Ah, this is interesting, because the field `pull_requests` has the content on my personal repo `gh api /repos/pzread/iree/actions/runs/3859339669`. Not very sure about the difference here.

Reply:
I removed the concurrency constraint on posting workflow since we don't have a reliable way to get the PR number here. We can use the PR commit SHA, but I don't think it's necessary since this workflow is small and if the main CI workflow is cancelled, this workflow won't find the succeeded benchmark job and skip the rest steps too.